modules/tensor_mechanics/src/materials/SingleVariableReturnMappingSolution.C

```diff
@@ -181,7 +181,10 @@ SingleVariableReturnMappingSolution::internalSolve(const Real effective_trial_st
                    max_permissible_scalar,
                    iter_output);
 
-    if (converged(residual, reference_residual))
+    if (converged(residual, reference_residual) ||
+        (scalar_lower_bound > 0 &&
+         (scalar_upper_bound - scalar_lower_bound) / (PETSC_MACHINE_EPSILON * scalar_lower_bound) <=
+             10.0))
```

Comment:
I had to adjust this check to get the Bison example to run through.  It seems like something along these lines is reasonable -- if the increment is small, we ought to be close enough.  The lower and upper bounds ensure that we are bracketing the solution, and if the lower and upper bounds are very close, then it ought to be close enough.